JWPlayer-Developer-Demo/ObjCPlayerViewController.m

```diff
@@ -103,9 +105,11 @@ - (void)createPlayer
     self.player.delegate = self;
     
     CGRect frame = self.view.bounds;
+
+    self.playerHeightConstraint.constant = frame.size.height / 2 - (44 + 64);
+    
     frame.origin.y = 64;
-    frame.size.height /= 2;
-    frame.size.height -= 44 + 64;
+    frame.size.height = self.playerHeightConstraint.constant;
```

Comment:
@kobertynska out of curiosity, why are we using `NSLayoutConstraint` instead of setting the player's height as a fraction of the parent view's height ? I am not familiar with `NSLayoutConstraint`, what are the benefits we are getting from this?

Reply:
@karimJWP we set the player's height as a fraction of the parent view's height, and we also set this height to playerHeightConstraint to avoid shifting of others views (buttons, callbacksView).
I've improved UI implementation in storyboard because the buttons were displayed under the player when I was testing the app.

Reply:
I see. I don't like storyboard changes because they are difficult to review; i didn't realize you were adding buttons until I checked out your branch and ran the app. I would say let's not add UI elements to this app. The purpose of this app is to be vanilla and to allow developers to use it to reproduce their bugs in an environment where we can isolate their issues and ensure the bugs are caused by our code and not theirs. 
I think making an app that shows how to link UI elements to our API could be valuable, but the place for that would be https://github.com/jwplayer/jwplayer-ios-bestPracticeApps. Notice the structure of the repo: different projects inherit from other projects to only show the relevant code. If you want you can create a  project that inherits from the base project and adds UI elements.

Reply:
@karimJWP I didn't add any new buttons and these buttons have been there for a while. There was a problem that these buttons were placed under the player, it seems like their position was related to screen size so sometimes they were hidden. So I did, changed hierarchy of existing UI elements to make them visible on the different screens. To sum up, I haven't introduced any new UI elements but made the existing ones to be placed properly. If you have any questions, please let me know.

Reply:
ah i see now I understand